Formula/s-nail.rb

```diff
@@ -0,0 +1,75 @@
+class SNail < Formula
+  desc "Fork of Heirloom mailx."
+  homepage "https://www.sdaoden.eu/code.html"
+  url "https://www.sdaoden.eu/downloads/s-nail-14.9.3.tar.gz"
+  sha256 "9048abe94c8b732ddefcd70b9f6052da16977de76b07598790e4c763e97f911d"
+
+  # Define options
+  option "with-maximal", "Anything on, including experimental features."
+  option "with-netsend", "Sending messages directly to the mail provider via the SMTP protocol, instead of requiring a local mail-transfer-agent (MTA) who does."
+  option "with-medium", "Possibly what people want who need nothing but a MIME-capable mailx(1)."
+  option "with-minimal", "This is the most plain mailx(1)-alike mode, but with MIME support."
+  option "with-nulli", "Anything that can be turned off is off but iconv(3)."
+  option "with-null", "Anything that can be turned off is off."
+
+  # Formula depends on OpenSSL per default and in MAXIMAL and NETSEND
+  depends_on "openssl" if build.with?("maximal") || build.with?("netsend") || (build.without?("medium") && build.without?("minimal") && build.without?("nulli") && build.without?("null"))
+  # Formula depends on IDNA per default and in MAXIMAL, NETSEND and MEDIUM
+  depends_on "libidn" if build.with?("maximal") || build.with?("netsend") || build.with?("medium") || (build.without?("minimal") && build.without?("nulli") && build.without?("null"))
+
+  def install
+    args = %W[
+      VAL_PREFIX=#{prefix}
+      VAL_PRIVSEP_USER=#{ENV["USER"]}
+    ]
+
+    # Make does not find the correct paths to OpenSSL itself
+    if build.with?("maximal") || build.with?("netsend")
+      args << "C_INCLUDE_PATH=#{Formula["openssl"].opt_include}"
+      args << "EXTRA_LDFLAGS=-L#{Formula["openssl"].opt_lib}"
+    elsif build.with?("medium") || build.with?("minimal") || build.with?("nulli") || build.with?("null")
+    else
+      args << "C_INCLUDE_PATH=#{Formula["openssl"].opt_include}"
+      args << "EXTRA_LDFLAGS=-L#{Formula["openssl"].opt_lib}"
+    end
+
+    # Determine config_preset
+    if build.with?("maximal")
+      args << "CONFIG=MAXIMAL"
+    elsif build.with?("netsend")
+      args << "CONFIG=NETSEND"
+    elsif build.with?("medium")
+      args << "CONFIG=MEDIUM"
+    elsif build.with?("minimal")
+      args << "CONFIG=MINIMAL"
+    elsif build.with?("nulli")
+      args << "CONFIG=NULLI"
+    elsif build.with?("null")
+      args << "CONFIG=NULL"
+    end
+
+    # Somehow make doesn't like a file called INSTALL being present.
+    # File is included in the .PHONY variable in the makefile but nonetheless.
+    # Install will fail if not renamed/deleted.
+    mv("INSTALL", "INSTALL.txt")
+    # Configure using above defined args and build
+    system "make", *args, "config"
+    # Build
+    system "make", "build"
+    # Install
+    system "make", "install"
+  end
+
+  def caveats; <<-EOS.undent
+    To use s-nail as mail client you need to set the correct permissions:
+
+    sudo chown root #{opt_libexec}/s-nail-privsep
```

Comment:
This will interfere with uninstall so is not something we can direct users to do. You'd need to find a way to relocate this directory outside the Cellar, possibly in `var`.